### PR TITLE
Updated move algorithm and Recipe auto put handling

### PR DESCRIPTION
**Pull request by MortenTobiasNielsen**
This has been a little long on the way, but I wanted to make sure that it worked as intended.

### CODE REVIEW

File: src/GUI/noname.cpp [hunk 1]
```diff
@@ -257,11 +257,11 @@ GUI_Base::GUI_Base( wxWindow* parent, wxWindowID id, const wxString& title, cons
 	wxBoxSizer* bSizer36;
 	bSizer36 = new wxBoxSizer( wxHORIZONTAL );
 
-	label_units = new wxStaticText( m_panel1, wxID_ANY, wxT("Units:"), wxDefaultPosition, wxSize( 60,-1 ), wxALIGN_RIGHT );
+	label_units = new wxStaticText( m_panel1, wxID_ANY, wxT("Amount:"), wxDefaultPosition, wxSize( 60,-1 ), wxALIGN_RIGHT );
 	label_units->Wrap( -1 );
 	bSizer36->Add( label_units, 0, wxALIGN_CENTER_VERTICAL|wxALL, 5 );
 
-	spin_units = new wxSpinCtrl( m_panel1, wxID_ANY, wxEmptyString, wxDefaultPosition, wxSize( 100,-1 ), wxSP_ARROW_KEYS, 0, 100000, 0 );
+	spin_units = new wxSpinCtrl( m_panel1, wxID_ANY, wxEmptyString, wxDefaultPosition, wxSize( 100,-1 ), wxALIGN_RIGHT|wxSP_ARROW_KEYS, 0, 100000, 0 );
```
Comment (theis999) [inline src/GUI/noname.cpp:264]: spin_units should be renamed to spin_amount

**Review by theis999 — COMMENTED** [review]
Codestyle-wise there is nothing wrong. My issue is most with intentions.
So I will need some clarification:

What is the intention of old_walking, is it just for backwards compatibility?
I don't understand how %.1f can be accurate enough for walking. Has it been tested that it is possible to walk between buildings?

I am not sure I like the idea of dropping items into buildings ( Z ), I just don't see the great benefit.
I think searching for back-to-front for ### start is the better approach. 
I don't have an opinion the addition of units on set recipe.
I can see that units was renamed to amount and then back units, which bloats the PR a little.

File: Lua Files/control.lua [hunk 2]
```diff
@@ -447,37 +439,40 @@ local function walk()
...
-			pos_pos = false
-			pos_neg = false
-			neg_pos = false
 		end
 	end
 end
 
+local function update_player_position()
+	player_position.x = tonumber(string.format("%.2f", player.position.x))
+	player_position.y = tonumber(string.format("%.2f", player.position.y))
```
[the remark this answers is not in the document]
Comment (MortenTobiasNielsen) [reply Lua Files/control.lua]: It seems like the smallest possible move is around 0.15, so I think two should be enough.

File: src/GenerateScript.cpp [hunk 3]
```diff
@@ -59,7 +59,17 @@ void GenerateScript::generate(wxWindow* parent, wxGrid* grid, dialog_progress_ba
...
 	size_t amount_of_tasks = steps.size();
 
-	for (int i = 0; i < amount_of_tasks; i++) {
+	size_t start_step = 0;
+
+	for (int i = amount_of_tasks - 1; i > 0; i--) {
+		extract_parameters(steps[i]);
+		if (task == "Start") {
+			start_step = i + 1;
+			break;
```
[the remark this answers is not in the document]
Comment (MortenTobiasNielsen) [reply src/GenerateScript.cpp:68]: It starts from the last step and stops at the first "Start" step (which is the last), so I think we agree. :D

File: src/GUI/noname.cpp [hunk 4]
```diff
@@ -257,11 +257,11 @@ GUI_Base::GUI_Base( wxWindow* parent, wxWindowID id, const wxString& title, cons
 	wxBoxSizer* bSizer36;
 	bSizer36 = new wxBoxSizer( wxHORIZONTAL );
 
-	label_units = new wxStaticText( m_panel1, wxID_ANY, wxT("Units:"), wxDefaultPosition, wxSize( 60,-1 ), wxALIGN_RIGHT );
+	label_units = new wxStaticText( m_panel1, wxID_ANY, wxT("Amount:"), wxDefaultPosition, wxSize( 60,-1 ), wxALIGN_RIGHT );
```
[the remark this answers is not in the document]
Comment (MortenTobiasNielsen) [reply src/GUI/noname.cpp:260]: I forgot to revert this change. Until we find a good way to separate amount (amount of buildings) in the grids. I don't know what to call amount of buildings in the grid, so I chose to have units, so it was possible to know what is what.

**Comment by MortenTobiasNielsen** [comment]
1. It is for backwards compatibility
2. I have made 1000's of steps with the new walking code and it works fine.
3. Agreed I need to refactor drop - I don't really see a use for drop so it is a little annoying to use time on
4. That is what the code is doing
5. It is very beneficial to have to possibility - I have used it 100's of times
6. Yeah I tried to revert it, but wasn't really successful

File: Lua Files/control.lua (hunk 2 above)
[the remark this answers is not in the document]
Comment (theis999) [reply Lua Files/control.lua]: It seems odd to me that the precision is so low, but alright.

File: src/GenerateScript.cpp [hunk 5]
```diff
@@ -974,22 +977,22 @@ void GenerateScript::row_filter(std::string step, std::string x_cord, std::strin
 	}
 }
 
-void GenerateScript::drop(std::string step, std::string action, std::string x_cord, std::string y_cord, std::string item) {
-	check_interact_distance(step, action, x_cord, y_cord, "Drop", "North");
+void GenerateScript::drop(std::string step, std::string action, std::string x_cord, std::string y_cord, std::string item, std::string building) {
+	check_interact_distance(step, action, x_cord, y_cord, building, "North");
```
[the remark this answers is not in the document]
Comment (MortenTobiasNielsen) [reply src/GenerateScript.cpp:999]: It doesn't work as intended currently and I am a little in doubt when it was actually broken. I have never actually used drop in a run. For now I won't use time solving the problem.

**Review by theis999 — APPROVED** [review]

**Comment by theis999** [comment]
That seems to answer my concerns.

I understand the need for drop in terms of completeness, but i doubt it is very useful for any TAS. So spend as little energy on it as you can.

We should make a note of removing old_walking in the future then.

**Pull request merged**